wp_api/src/request.rs

```diff
@@ -439,6 +449,175 @@ impl WpNetworkResponse {
     {
         parser(self)
     }
+
+    pub fn is_rate_limit_exceeded(&self) -> bool {
+        self.status_code == 429
+    }
+
+    pub fn get_retry_after(&self) -> Option<u64> {
+        self.header_map
+            .inner
+            .get(http::header::RETRY_AFTER)
+            .and_then(|h_v| h_v.to_str().ok())
+            .and_then(|s| HttpRetryDuration::from_str(s).ok())
+            .map(|d| d.seconds)
+    }
+
+    pub fn is_http_authentication_required(&self) -> bool {
+        self.status_code == 401 || self.status_code == 403
+    }
+
+    pub fn get_http_auth_method(
+        &self,
+    ) -> Result<Option<HttpAuthMethod>, HttpAuthMethodParsingError> {
```

Comment:
`Option<Result<HttpAuthMethod, HttpAuthMethodParsingError>>` might make more sense? `None` means the header does not exists and `Some(result)` means we tried to parse the header. Or, just return `Result<...>` type by adding a "missing header" to the error?